crates/polars-ops/src/series/ops/cum_agg.rs

```diff
@@ -216,15 +216,37 @@ pub fn cum_max(s: &Series, reverse: bool) -> PolarsResult<Series> {
 }
 
 pub fn cum_count(s: &Series, reverse: bool) -> PolarsResult<Series> {
-    if reverse {
-        let ca: NoNull<UInt32Chunked> = (0u32..s.len() as u32).rev().collect();
-        let mut ca = ca.into_inner();
-        ca.rename(s.name());
-        Ok(ca.into_series())
-    } else {
-        let ca: NoNull<UInt32Chunked> = (0u32..s.len() as u32).collect();
-        let mut ca = ca.into_inner();
-        ca.rename(s.name());
-        Ok(ca.into_series())
+    // Fast paths for no nulls
+    if s.null_count() == 0 {
+        let out = cum_count_no_nulls(s.name(), s.len(), reverse);
+        return Ok(out);
     }
+
+    let mut count = 0 as IdxSize;
+    let f = |v: bool| {
+        if v {
+            count += 1
+        }
+        count
+    };
+
+    let ca = s.is_not_null();
+    let iter = ca.into_no_null_iter();
+    let out: NoNull<IdxCa> = match reverse {
+        false => iter.map(f).collect(),
+        true => iter.rev().map(f).collect_reversed(),
+    };
+    Ok(out.into_inner().into())
+}
+
+fn cum_count_no_nulls(name: &str, len: usize, reverse: bool) -> Series {
+    let start = 1 as IdxSize;
+    let end = len as IdxSize + 1;
+    let ca: NoNull<IdxCa> = match reverse {
```

Comment:
Nit: can we `if else` on `boolean`s?

Reply:
I normally would but everywhere else in this file the `reverse` is matched. I'll update 👍